src/wallet/interfaces.cpp

```diff
@@ -410,6 +410,11 @@ class WalletImpl : public Wallet
             result.unconfirmed_watch_only_balance = bal.m_watchonly_untrusted_pending;
             result.immature_watch_only_balance = bal.m_watchonly_immature;
         }
+        if (m_wallet->IsWalletFlagSet(WALLET_FLAG_AVOID_REUSE)) {
+            const auto full_bal = GetBalance(*m_wallet, 0, false);
```

Comment:
GetBalance receives `bool avoid_reuse` in the last argument, should not be `true` or just remove the if condition directly and pass the result of `IsWalletFlagSet(WALLET_FLAG_AVOID_REUSE)` entirely.
```suggestion
            const auto full_bal = GetBalance(*m_wallet, 0, true);
```

Reply:
That parameter is slightly misleading. When set to `true`, it will ignore the used balance, so it needs to be `false` here in order to get the full balance to calculate the used balance. This is what `getbalances` does.

However, I think it would be better to move all of this into `GetBalance` itself and just have it always compute the used balance for us rather than having the caller do this extra computation.

Reply:
Sorry, it's clear, I was incorrect in my statement, but what I meant is we could use the flag instead of `false` (e.g.: `GetBalance(*m_wallet, 0, !(m_wallet->IsWalletFlagSet(WALLET_FLAG_AVOID_REUSE)));`) and remove the `if`.

I agree with moving this piece into `GetBalance`, the other only place where this is called with this argument is in `coins.cpp`.